### PR TITLE
Tween property implicit type conversion

### PR DESCRIPTION
**Pull request by DavidSichma**
Addresses #53879
Issue is also about how to make such errors more visible. This just converts types.

--------

This performs type conversion inside Tween::tween_property and PropertyTweener::from if necessary.
These are the same conversion that GdScript applies to typed function arguments and should be the least surprising.
Note: GdScript conversions are not that strict and allow bool -> float among others. 

Also added a check if the property exists in the first place.

Conversion happens without DEBUG_ENABLED because Tween-internals rely on matching types.

### CODE REVIEW

File: scene/animation/tween.cpp [hunk 1]
```diff
@@ -75,12 +75,17 @@ Ref<PropertyTweener> Tween::tween_property(Object *p_target, NodePath p_property
 	ERR_FAIL_COND_V_MSG(!valid, nullptr, "Tween invalid. Either finished or created outside scene tree.");
 	ERR_FAIL_COND_V_MSG(started, nullptr, "Can't append to a Tween that has started. Use stop() first.");
 
-#ifdef DEBUG_ENABLED
-	Variant::Type property_type = p_target->get_indexed(p_property.get_as_property_path().get_subnames()).get_type();
-	ERR_FAIL_COND_V_MSG(property_type != p_to.get_type(), Ref<PropertyTweener>(), "Type mismatch between property and final value: " + Variant::get_type_name(property_type) + " and " + Variant::get_type_name(p_to.get_type()));
-#endif
+	bool access_ok;
+	Variant::Type target_type = p_target->get_indexed(p_property.get_as_property_path().get_subnames(), &access_ok).get_type();
+	ERR_FAIL_COND_V_MSG(!access_ok, nullptr, "Can't access property " + p_property);
```
Comment (KoBeWi) [inline scene/animation/tween.cpp:80]: The message could be reworded and include more details, like "Can't find property 'property' in object 'object'.

File: scene/animation/tween.cpp [hunk 2]
```diff
@@ -75,12 +75,17 @@ Ref<PropertyTweener> Tween::tween_property(Object *p_target, NodePath p_property
...
 
-	Ref<PropertyTweener> tweener = memnew(PropertyTweener(p_target, p_property, p_to, p_duration));
+	Variant to_value;
+	const Variant *src = &p_to;
+	Callable::CallError ce;
+	Variant::construct(target_type, to_value, &src, 1, ce);
+	ERR_FAIL_COND_V_MSG(ce.error != Callable::CallError::CALL_OK, nullptr, "Can't convert final value to target type: " + Variant::get_type_name(p_to.get_type()) + " to " + Variant::get_type_name(target_type));
```
Comment (KoBeWi) [inline scene/animation/tween.cpp:86]: You could leave the old message here.

File: scene/animation/tween.cpp [hunk 3]
```diff
@@ -699,7 +709,9 @@ void PropertyTweener::start() {
 	}
 
 	if (do_continue) {
-		initial_val = target_instance->get_indexed(property);
+		bool access_ok;
+		initial_val = target_instance->get_indexed(property, &access_ok);
+		ERR_FAIL_COND_MSG(!access_ok, "Target property not accessible, aborting Tweener.");
```
Comment (KoBeWi) [inline scene/animation/tween.cpp:714]: This is already checked when creating the Tweener (you can't change the property), so no need to check it again.

File: scene/animation/tween.cpp [hunk 4]
```diff
@@ -658,7 +663,12 @@ void Tween::_bind_methods() {
 }
 
 Ref<PropertyTweener> PropertyTweener::from(Variant p_value) {
-	initial_val = p_value;
+	Variant::Type target_type = initial_val.get_type();
+	const Variant *src = &p_value;
+	Callable::CallError ce;
+	Variant::construct(target_type, initial_val, &src, 1, ce);
+	ERR_FAIL_COND_V_MSG(ce.error != Callable::CallError::CALL_OK, this, "Can't convert start value to target type: " + Variant::get_type_name(p_value.get_type()) + " to " + Variant::get_type_name(target_type));
```
Comment (KoBeWi) [inline scene/animation/tween.cpp:670]:
```suggestion
	ERR_FAIL_COND_V_MSG(ce.error != Callable::CallError::CALL_OK, this, "Type mismatch between property and new initial value: " + Variant::get_type_name(p_value.get_type()) + " and " + Variant::get_type_name(target_type));
```

**Comment by KoBeWi** [comment]
`tween_method()` could get a type check/conversion too.

**Comment by akien-mga** [comment]
Superseded by #65072.

**Pull request closed without merging**